doc/tcti-spi.md

```diff
@@ -0,0 +1,250 @@
+{
+    // Cast our user data
+    PLATFORM_USERDATA* platform_data = (PLATFORM_USERDATA*) user_data;
+
+    // Release CS and release the bus for other devices
+    gpio_set_level(platform_data->cs_pin, 1);
+    spi_device_release_bus(platform_data->spi);
+
+    return TSS2_RC_SUCCESS;
+}
+
+// Transfer cnt bytes from data_out to device and read response into data_in
+TSS2_RC platform_spi_transfer (void* user_data, const void *data_out, void *data_in, size_t cnt)
+{
+    // Cast our user data
+    PLATFORM_USERDATA* platform_data = (PLATFORM_USERDATA*) user_data;
+
+    // Maximum transfer size is 64 byte because we don't use DMA (and the TPM doesn't support more anyway)
+    if (cnt > 64) {
+        return TSS2_TCTI_RC_BAD_VALUE;
+    }
+
+    // At least one of the buffers has to be set
+    if (data_out == NULL && data_in == NULL) {
+        return TSS2_TCTI_RC_BAD_VALUE;
+    }
+
+    // Clear receive buffer
+    if (data_in != NULL) {
+        memset(data_in, 0, cnt);
+    }
+
+    // Setup transaction
+    spi_transaction_t t;
+    memset(&t, 0, sizeof(t));
+    t.length = cnt*8;
+    t.tx_buffer = data_out;
+    t.rx_buffer = data_in;
+
+    // Transmit
+    esp_err_t ret = spi_device_polling_transmit(platform_data->spi, &t);
+    if (ret != ESP_OK) {
+        return TSS2_TCTI_RC_IO_ERROR;
+    }
+
+    return TSS2_RC_SUCCESS;
+}
+
+// Sleeps for the specified amount of milliseconds
+void platform_sleep_ms (void* user_data, int32_t milliseconds)
+{
+    // Sleep the specified amount of milliseconds, but at least one tick to feed the watchdog
+    TickType_t sleep_ticks = milliseconds / portTICK_PERIOD_MS;
+    if (sleep_ticks == 0) {
+        sleep_ticks = 1;
+    }
+    vTaskDelay(sleep_ticks);
+}
+
+// Start a timeout timer with specified expiry in milliseconds
+void platform_start_timeout (void* user_data, int32_t milliseconds)
+{
+    // Cast our user data
+    PLATFORM_USERDATA* platform_data = (PLATFORM_USERDATA*) user_data;
+
+    // Store timeout expiry time
+    platform_data->timeout_expiry = esp_timer_get_time() + (milliseconds*1000);
+}
+
+// Check if the timeout timer started by platform_start_timeout has already expired
+bool platform_timeout_expired (void* user_data)
+{
+    // Cast our user data
+    PLATFORM_USERDATA* platform_data = (PLATFORM_USERDATA*) user_data;
+
+    // Check if timeout already expired
+    return esp_timer_get_time() < platform_data->timeout_expiry;
+}
+
+// Frees the platform user_data struct and everything inside
+void platform_finalize(void* user_data)
+{
+    // Cast our user data
+    PLATFORM_USERDATA* platform_data = (PLATFORM_USERDATA*) user_data;
+
+    // Free resources inside user_data like SPI device handles here
+    // ...
+    
+    // Free user_data
+    free(platform_data);
+}
+
+// Creates a new platform struct with the given SPI device handle and chip select
+TSS2_TCTI_SPI_PLATFORM create_tcti_spi_platform(spi_device_handle_t spi, gpio_num_t cs_pin)
```

Comment:
returning a full struct here makes it impossible to deal with any errors - e.g. if malloc fails.
For other tctis that use this as a reference, the create_tcti_spi_platform might be less trivial.
Probably makes sense to return a pointer instead.